pageserver/src/tenant.rs

```diff
@@ -1366,6 +1338,36 @@ impl Tenant {
         Ok(timeline_preloads)
     }
 
+    fn load_timeline_metadata(
```

Comment:
Noting this is not `async fn` for some reason I cannot see?

Reply:
it's not `async fn` for lifetime improvements. That way, we can pass in a `&Arc<Tenant>` instead of having to clone the Arc. It's not a very often run operation but the fix to make it non-async fn isn't complicated either.